decorator.py

```diff
@@ -0,0 +1,31 @@
+def type_check(correct_type):
+    """
+    decorator function, checks if the variable is of the required type
+    :param correct_type: required type of variable
+    :return: fun with var or a message if the variable is not of the appropriate type
+    """
+    def decorator(fun):
+        def inner(var):
+            if correct_type == type(var):
+                return fun(var)
+            return "No access to the function"
+        return inner
+    return decorator
+
+
+@type_check(int)
+def times2(num):
+    return num * 2
+
+
+@type_check(str)
+def times1(st):
```

Comment:
replace 'st' Variable names - https://workat.tech/machine-coding/tutorial/writing-meaningful-variable-names-clean-code-za4m83tiesy0 - variable names, abbreviation reference is missing

Reply:
I changed it to string